www/js/lib/cookies.js

```diff
@@ -21,11 +21,43 @@ define([], function() {
 |*|
 \*/
 
+var storeType = testStorageSupport();
+
+// Tests for localStorage support
+function testStorageSupport() {
+  // Prefer localStorage if supported due to some platforms blocking cookies in local contexts
+  // DEV: In FF extensions, cookies are blocked since at least FF 68.6 but possibly since FF 55 [kiwix-js #612]
+  var type = 'local_storage';
+  // First test for localStorage support
+  var localStorageTest = false;
+  try {
+    localStorageTest = 'localStorage' in window && window['localStorage'] !== null;
+    // DEV: Above test returns true in IE11 running from file:// protocol, but attempting to write a key to
+    // localStorage causes an exception; so to test fully, we must now attempt to write and remove a test key
+    if (localStorageTest) {
+      localStorage.setItem('kiwixStorage', '');
+      localStorage.removeItem('kiwixStorage');
+    }
+  } catch (e) {
+    localStorageTest = false;
+    console.log('LocalStorage is not supported!');
+  }
+  // Fall back to cookie if localStorage fails
+  if (!localStorageTest) type = 'cookie';
+  console.log('Storage test: type: ' + type);
+  return type;
+}
+
 var docCookies = {
```

Comment:
I think this variable and the file itself (with its header) should be renamed to something more generic. Maybe `settingsStorage`?

Reply:
@mossroy I can do this, but let me check that you mean that throughout the app where "cookies" are set, we should in now use, instead of `cookies.getItem('mySetting')`, something like `settingsStorage.getItem('mySetting')`?

Reply:
Note to self: in doing this, I think we should include #438 (i.e. remove deprecated `escape` and `unescape`, using `encodeURIComponent` etc. instead).

Reply:
Yes, when we mention setting/reading cookies in the app, we should instead talk about setting/reading some stored settings.
And yes, it would be the right time to implement #438, I had forgotten this one

Reply:
I've added the latest Mozilla framework code (albeit from 2014) that fixe #438.

Reply:
Done.